### ISSUE TITLE
Unable to configure new dashboard widget

### ISSUE THREAD

**Issue by shaunmarx** [OctopusDeploy/OctoTFS#142]
# Prerequisites

- [x] I have verified the problem exists in the latest version
- [x] I have searched [open](https://github.com/OctopusDeploy/Issues/issues) and [closed](https://github.com/OctopusDeploy/Issues/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed) issues to make sure it isn't already reported
- [x] I have written a descriptive issue title
- [x] I have linked the original source of this report
- [x] I have tagged the issue appropriately (area/*, kind/bug, tag/regression?)

# The bug

A new octopus status widget can not be added to a dashboard as configuring the widget errors.

## What I expected to happen

Configuration is shown and allows you to configure the dashboard widget after which it is added to the dashboard.

## Steps to reproduce

1. Go to an Azure Devops instance
2. Navigate to the Dashboard
3. Edit the dashboard
4. Add an octopus status widget
5. Notice the blank configuration when attempting to edit the widget

### Screen capture
![image](https://user-images.githubusercontent.com/6700195/57004043-c01d5300-6c0e-11e9-8b58-7bb5ac5a7927.png)

### Log exerpt

```
Must use title or breadcrumb items + headerItemPicker
VM83 VSTS.Extensions.TestProject:1 Uncaught (in promise) ObjectisCanceled: true__proto__: Object
vss-bundle-common-v6EqIrreGtxkzLZ8M7mUsRqA447RJJvZKiDfRKOwiGiI=:95 

Rejected XDM promise with no reject callbacks
n._reject @ vss-bundle-common-v6EqIrreGtxkzLZ8M7mUsRqA447RJJvZKiDfRKOwiGiI=:95
vss-bundle-common-v6EqIrreGtxkzLZ8M7mUsRqA447RJJvZKiDfRKOwiGiI=:95 

TypeError: Cannot read property 'statusType' of undefined
    at vss-bundle-async-vkXP3LBaTIkIXt3QhzEOCN9Fikmu3F-mKkd5k1qO8Xm8=:28
    at n._wrapCallback (vss-bundle-common-v6EqIrreGtxkzLZ8M7mUsRqA447RJJvZKiDfRKOwiGiI=:95)
    at Array.<anonymous> (vss-bundle-common-v6EqIrreGtxkzLZ8M7mUsRqA447RJJvZKiDfRKOwiGiI=:95)
    at vss-bundle-common-v6EqIrreGtxkzLZ8M7mUsRqA447RJJvZKiDfRKOwiGiI=:95
n._reject @ vss-bundle-common-v6EqIrreGtxkzLZ8M7mUsRqA447RJJvZKiDfRKOwiGiI=:95
octo-status.js:61 

Failed to query service connection API: 'https://deploy.octopushq.com/api/dashboard?projectId=null'. Status Code: 'InternalServerError', Response from server: '{
  "ErrorMessage": "Object reference not set to an instance of an object."
}'
(anonymous) @ octo-status.js:61
```

## Links
https://help.octopus.com/t/cannot-configure-azure-devops-dashboard-widget/22808

**Comment by shaunmarx** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Unable to configure ADO Extension widget](https://trello.com/c/qRi9NXJj/757-unable-to-configure-ado-extension-widget)

**Comment by captaintightpants** [comment]
I'm hitting the same issue. Looks like it's stuck here 

`var __extends=this&&this.__extends||function(){var n=function(t,i){return n=Object.setPrototypeOf||{__proto__:[]}instanceof Array&&function(n,t){n.__proto__=t}||function(n,t){for(var i in t)t.hasOwnProperty(i)&&(n[i]=t[i])},n(t,i)};return function(t,i){function r(){this.constructor=t}n(t,i);t.prototype=i===null?Object.create(i):(r.prototype=i.prototype,new r)}}(),__assign=this&&this.__assign||function(){return __assign=Object.assign||function(n){for(var t,r,i=1,u=arguments.length;i<u;i++){t=arguments[i];for(r in t)Object.prototype.hasOwnProperty.call(t,r)&&(n[r]=t[r])}return n},__assign.apply(this,arguments)};define("Dashboards/Components/LegacyConfigHostAdapter",["require","exports","react","react-dom","q","Dashboards/Scripts/Resources/TFS.Resources.Dashboards","VSS/LWP","Dashboards/WidgetContracts/Internal/ConfigProps","Dashboards/Scripts/Common","TFS/Dashboards/WidgetContracts","VSS/Contributions/Controls","./LegacyHostUtilities","TFS/Dashboards/WidgetHelpers","VSS/Utils/String"],function(n,t,i,r,u,f,e,o,s,h,c,l,a,v){function p(n,t,i){var r=l.isFirstPartyContribution(t),u=null,f;return r||(u=l.extractRelativeContributionId(i.config.configurationContributionId)),f=r?__assign({},i.legacyRuntimeProps,{widgetTypeId:i.config.contributionId}):null,c.createContributedControl(n,t,f,null,u)}function w(n){return{name:n.config.name,customSettings:{data:n.config.settings,version:n.config.settingsVersion},size:n.config.size}}Object.defineProperty(t,"__esModule",{value:!0});var y=function(n){function t(t){var i=__construct(this,n,[t]);for(var r in i)i.hasOwnProperty(r)&&(this[r]=i[r]);return i.notify=function(n,t){var o=t&&t.data,i,r,e;if(n===a.WidgetEvent.ConfigurationChange){i=o;r=Object.assign({},this.props.config,{settings:i.data,settingsVersion:i.version});this.props.parentProps.onUpdate(r)}else throw new Error(f.ConfigHostAdapter_UnsupportedAPI);return e={getResponse:undefined},u(e)},i}return __extends(t,n),t.prototype.render=function(){return i.createElement("div",{className:"classic-config-container"})},t.prototype.componentDidUpdate=function(n){var t=this;if(n.saveValidationTimeStamp!==this.props.saveValidationTimeStamp)if(this.contributedControl.onSave()!=null)this.contributedControl.onSave().then(function(n){var i=Object.assign({},t.props.config,{settings:n.customSettings.data,settingsVersion:n.customSettings.version||t.props.config.settingsVersion});t.props.parentProps.onValidationResponse(i,n.isValid)},function(){t.props.parentProps.onValidationResponse(t.props.config,!1)});else this.props.parentProps.onValidationResponse(this.props.config,!1);this.notifyCustomConfigOfGeneralSettingsChange(n)},t.prototype.componentDidMount=function(){var n=this,t=r.findDOMNode(this),i=$(t);this.props!=null&&l.loadContribution(this.props.config.configurationContributionId).then(function(t){p(i,t,n.props).then(function(t){n.contributedControl=t;var i=w(n.props),r={notify:n.notify.bind(n)};t.load(i,r).then(function(t){t.statusType!==h.WidgetStatusType.Success&&n.passErrorToHost(f.ConfigHostAdapter_LoadFailure,"VS403685",t.state);n.props.parentProps.onConfigLoaded(l.interpretStatus(t))},function(t){n.passErrorToHost(f.ConfigHostAdapter_LoadFailure,"VS403684",t)})},function(t){n.passErrorToHost(f.ConfigHostAdapter_ContributionLoadError,"VS403683",t)})},function(t){n.passErrorToHost(f.ConfigHostAdapter_ContributionLoadError,"VS403682",t)})},t.prototype.notifyCustomConfigOfGeneralSettingsChange=function(n){if(n.config.size.columnSpan!=this.props.config.size.columnSpan||n.config.size.rowSpan!=this.props.config.size.rowSpan||n.config.name!=this.props.config.name){var t={data:{WidgetName:this.props.config.name,WidgetSize:{columnSpan:this.props.config.size.columnSpan,rowSpan:this.props.config.size.rowSpan}}};this.contributedControl.listen&&this.contributedControl.listen(a.WidgetEvent.GeneralSettingsChanged,t)}},t.prototype.passErrorToHost=function(n,t,i){var r=s.ErrorMessageHelper.parseErrorToString(i);this.props.parentProps.onConfigLoaded({outcome:o.LoadingStatus.failure,outcomeDescription:v.format(n,t,r)})},t.componentType="LegacyConfigHostAdapter",t}(i.Component);t.LegacyConfigHostAdapter=y;e.registerLWPComponent(y.componentType,y);t.loadConfig=p;t.packLegacySettings=w});`

**Comment by captaintightpants** [comment]
Somehow statusType does not get set on the object 
`t.load(i,r).then(function(t){t.statusType!==h.WidgetStatusType.Success&&n.passErrorToHost(f.ConfigHostAdapter_LoadFailure,"VS403685",t.state);n.props.parentProps.onConfigLoaded(l.interpretStatus(t))}`

**Comment by thedewi** [comment]
It looks like this wasn't completely fixed in #137, but I'm hopeful it has been after #154, shipped in version 4.0.375 today.
